engine/skiplist.cpp

```diff
@@ -60,17 +60,28 @@ void SkiplistNode::SeekNode(const pmem::obj::string_view &key,
       if (next_next.GetTag()) {
         if (prev->CASNext(i, next, next_next.RawPointer())) {
           if (--next->valid_links == 0) {
-            if (to_delete == nullptr) {
-              to_delete.reset(new std::vector<SkiplistNode *>());
-            }
-            to_delete->push_back(next.RawPointer());
+            to_delete.push_back(next.RawPointer());
           }
         }
         // if prev is marked deleted before cas, cas will be failed, and prev
         // will be roll back in next round
         continue;
       }
-      int cmp = compare_string_view(key, next->UserKey());
+
+      DLRecord *next_pmem_record = next->record;
+      // pmem record maybe updated during seek, then the next record could be
+      // invalid, so we need to catch exceptions here
+      int cmp;
+      try {
+        cmp = compare_string_view(key, next->UserKey());
+      } catch (std::runtime_error &) {
+        continue;
+      }
+      // pmem record maybe updated before comparing string, then the compare
```

Comment:
before? or after? if before, maybe move  line 75 ~ line 79 to after‘ `double check`

Reply:
before or during, so we need to double check after compare, to ensure it's never changed